src/IQFeed.CSharpApiClient/Lookup/Historical/Messages/TickMessage.cs

```diff
@@ -11,7 +11,8 @@ public class TickMessage : ITickMessage
         public const string TickDateTimeFormat = "yyyy-MM-dd HH:mm:ss.ffffff";
 
         public TickMessage(DateTime timestamp, double last, int lastSize, int totalVolume, double bid, double ask,
-            long tickId, char basisForLast, int tradeMarketCenter, string tradeConditions, string requestId = null)
+            long tickId, char basisForLast, int tradeMarketCenter, string tradeConditions, int tradeAggressor,
```

Comment:
Again this comes back to whether we want to support multiple protocols in the same release.

Reply:
not for live data but lets consider my situation. I collected bunch of data over the years lets say with protocol 6.0. I need a way to parse my existing data with newest client version. This is easily achievable though

Reply:
So as I said above, will this all be via the HistorialFileFacade, or is there another entry-point? 
I think the right approach would be within the message parser for TickMessage, to simply check the length of the values array which is parsed from the message. If it is long enough to have the two additional fields, we take that path, and if now, we parse with default values for those two fields.
All parsing requests should go though those methods shouldn't they?
If you like that approach, I'll implement tomorrow and add a couple of tests with the old test data as well as the new.

Reply:
Yes I was also thinking about this approach. I think it makes total sense. From what I see, I think we just need to adapt ParseFromFile from TickMessage. Others are just used from the live engine so we dont need to do anything.

Thank you again for your effort in all of this. This will help a lot of other users!

Reply:
OK, I've done as we've agreed, and that's working fine.
To be honest, I did most of this work a year ago, and I've been using my version (with my version of Dynamic fields) ever since, so just happy we can make it available to everyone else.